--- /c/Users/vasum/VB_champsim_Optimised_Hermesed/champsim_v17/inc/ooo_cpu.h	2026-06-13 04:09:56.426558300 +0900
+++ /c/Users/vasum/VB_champsim_Optimised_Hermesed/champsim_v17_B8/inc/ooo_cpu.h	2026-06-19 02:15:44.058075800 +0900
@@ -104,7 +104,7 @@
                 }
             }
         }
-        if (tombstone_count > 128) compact();
+        if (tombstone_count > 32) compact();
     }
 
     void compact() {
